src/interpreter/Environment.ts

```diff
@@ -152,7 +157,11 @@ export class Environment {
         );
 
         if (source) {
-            return source.get(lowercaseName)!;
+            let variableToReturn = source.get(lowercaseName)!;
+            if (this.isMockedFunction(variableToReturn, lowercaseName)) {
```

Comment:
Nit, but this calls getMockFunction() twice.  Duplicate calls have led to misc. bugs before so is there a way to shorten this?  Like:

    let mockedFunc = getMockFunction(...)
    if (mockedFunc) then return mockedFunc else return source.get(lowercaseName)

Reply:
I think we probably could clean this up to prevent duplicate calls I just really wanted to make sure that the variable was a `Callable` so that we didn't return a mocked function for some `boolean` variable for example.

I'll take a look

Reply:
How would you feel about using `this.mockFunctions.has(possibleMockName)` inside of `isMockedFunction`?

Reply:
Sounds good to me.